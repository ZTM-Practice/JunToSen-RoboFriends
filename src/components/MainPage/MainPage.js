import React, { useEffect } from 'react';
import CardList from '../CardList/CardList';
import SearchBox from '../SearchBox/SearchBox';
import Scroll from '../Scroll/Scroll';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import './MainPage.css';

const MainPage = (props) => {
    const { searchField, isPending, robots, onRequestRobots, onSearchChange } = props;

    useEffect(() => {
        onRequestRobots();
    }, []);

    const filteredRobots = robots.filter((robot) => {
        return robot.name.toLowerCase().includes(searchField.toLowerCase())
    })

    return isPending ?
        <h1>LOADING</h1> :
        (
            <div className='tc'>
                <h1 className='f1'>RoboFriends</h1>
                <SearchBox searchChange={onSearchChange}/>
                <Scroll>
                    <ErrorBoundary>
                        <CardList id="robotList" robots={filteredRobots} />
                    </ErrorBoundary>
                </Scroll>
            </div>
        )
}

export default MainPage;
