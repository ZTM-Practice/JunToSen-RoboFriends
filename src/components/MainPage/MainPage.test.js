import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

describe('MainPage Component', () => {
    it('renders MainPage Component', () => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            onSearchChange: jest.fn(),
            robots: [],
            searchField: '',
            isPending: false,
        };
        expect(shallow(<MainPage {...mockProps}/>)).toMatchSnapshot();
    });

    it('filters robots correctly 1', () => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            onSearchChange: jest.fn(),
            robots: [{
              id: 1,
              name: 'Leanne Graham',
              username: 'Bret',
              email: 'Sincere@april.biz'
            }],
            searchField: 'Leanne',
            isPending: false
          };
        const wrapper = shallow(<MainPage {...mockProps}/>);
        expect(wrapper.find('#robotList').prop('robots').length).toEqual(1);    
    });

    it('filters robots correctly 2', () => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            onSearchChange: jest.fn(),
            robots: [{
              id: 1,
              name: 'Leanne Graham',
              username: 'Bret',
              email: 'Sincere@april.biz'
            }],
            searchField: 'John',
            isPending: false
          };
          const wrapper = shallow(<MainPage {...mockProps}/>);
          expect(wrapper.find('#robotList').prop('robots').length).toEqual(0);    
        })
});
