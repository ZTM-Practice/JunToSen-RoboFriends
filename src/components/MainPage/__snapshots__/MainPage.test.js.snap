// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component renders MainPage Component 1`] = `ShallowWrapper {}`;
